sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuShuffledSizedHashJoinExec.scala

```diff
@@ -826,35 +838,44 @@ object GpuShuffledAsymmetricHashJoinExec {
       val exprs = BoundJoinExprs.bind(joinType, leftKeys, leftOutput, rightKeys,
         rightOutput, condition, buildSide)
       val buildQueue = mutable.Queue.empty[T]
-      val (buildRows, buildSize) = closeOnExcept(buildQueue) { _ =>
+      val (mayTruncatedBuildRows, mayTruncatedBuildSize) = closeOnExcept(buildQueue) { _ =>
         fetchProbeTargetSize(probeBuildIter, buildQueue, gpuBatchSizeBytes)
       }
       val baseBuildIter = setupForJoin(buildQueue, rawBuildIter, exprs.buildTypes,
         gpuBatchSizeBytes, metrics)
-      if (buildRows <= Int.MaxValue && buildSize <= gpuBatchSizeBytes) {
+      if (mayTruncatedBuildRows <= Int.MaxValue && mayTruncatedBuildSize <= gpuBatchSizeBytes) {
         assert(!probeBuildIter.hasNext, "build side not exhausted")
         getJoinInfoSmallBuildSide(joinType, buildSide, condition, exprs,
-          baseBuildIter, buildRows, buildSize,
+          baseBuildIter, mayTruncatedBuildRows, mayTruncatedBuildSize,
           rawStreamIter, gpuBatchSizeBytes, metrics)
       } else {
         // The natural build side does not fit in a single batch, so use the stream side
         // as the hash table if we can fit it in a single batch.
         val streamQueue = mutable.Queue.empty[T]
-        val (streamRows, streamSize) = closeOnExcept(streamQueue) { _ =>
+        val (mayTruncatedStreamRows, mayTruncatedStreamSize) = closeOnExcept(streamQueue) { _ =>
           fetchProbeTargetSize(probeStreamIter, streamQueue, gpuBatchSizeBytes)
         }
         val streamIter = setupForJoin(streamQueue, rawStreamIter, exprs.streamTypes,
           gpuBatchSizeBytes, metrics)
-        if (streamRows <= Int.MaxValue && streamSize <= gpuBatchSizeBytes) {
+        if (mayTruncatedStreamRows <= Int.MaxValue && mayTruncatedStreamSize <= gpuBatchSizeBytes) {
           assert(!probeStreamIter.hasNext, "stream side not exhausted")
```

Comment:
I think if we have a stream side that is exactly `gpuBatchSizeBytes` or exactly `Int.MaxValue` then this assert will trigger. Not specific to your PR, but the question is, is that possible, and what happens in that case? It seems our metric (`BUILD_DATA_SIZE`) would be wrong, at least.